src/main/java/gregtech/client/model/modelfactories/ActiveVariantBlockBakedModel.java

```diff
@@ -1,126 +1,118 @@
-            activeState = ((IExtendedBlockState) state).getValue(VariantActiveBlock.ACTIVE);
-        }
+        if (state == null) return Collections.emptyList();
 
-        ResourceLocation registryName = state.getBlock().getRegistryName();
-        if (registryName == null) {
-            throw new IllegalArgumentException("Block " + state.getBlock().getTranslationKey() + " has null registry name");
-        }
-        if (activeState) {
-            mrl = new ModelResourceLocation(registryName,
-                    "active=true,variant=" + state.getProperties().entrySet().stream().filter(p -> p.getKey().getName().equals("variant")).map(e -> {
-                        IProperty<?> p = e.getKey();
-                        return getPropertyName(p, e.getValue());
-                    }).collect(Collectors.joining()));
-        } else {
-            mrl = new ModelResourceLocation(registryName,
-                    "active=false,variant=" + state.getProperties().entrySet().stream().filter(p -> p.getKey().getName().equals("variant")).map(e -> {
-                        IProperty<?> p = e.getKey();
-                        return getPropertyName(p, e.getValue());
-                    }).collect(Collectors.joining()));
-        }
-        IBakedModel m = Minecraft.getMinecraft().blockRenderDispatcher.getBlockModelShapes().getModelManager().getModel(mrl);
-        TextureAtlasSprite textureAtlasSprite = m.getParticleTexture();
-        particle.set(textureAtlasSprite);
-        if (MinecraftForgeClient.getRenderLayer() == BloomEffectUtil.getRealBloomLayer()) {
-            if (ConfigHolder.client.casingsActiveEmissiveTextures) {
-                quads = new ArrayList<>();
-                for (BakedQuad b : m.getQuads(state, side, rand)) {
-                    if (b.getSprite().getIconName().contains("bloom")) {
-                        quads.add(b);
-                    }
-                }
-            }
-            else {
-                quads = Collections.emptyList();
-            }
+        IBakedModel m = getModel(state);
+
+        // If bloom is enabled, bloom textures are rendered on bloom layer
+        // If bloom is disabled (either by model specific bloom config or the presence of O**ifine shaders)
+        // it is rendered on CUTOUT layer instead.
+        if (getBloomConfig()) {
+            return MinecraftForgeClient.getRenderLayer() != BloomEffectUtil.BLOOM ?
+                    m.getQuads(state, side, rand) :
+                    m.getQuads(state, side, rand).stream()
+                            .filter(q -> q.getSprite().getIconName().contains("bloom"))
+                            .collect(Collectors.toList());
         } else {
-            quads = new ArrayList<>(m.getQuads(state, side, rand));
+            if (MinecraftForgeClient.getRenderLayer() == BloomEffectUtil.BLOOM) {
+                return Collections.emptyList();
+            } else if (MinecraftForgeClient.getRenderLayer() == BlockRenderLayer.CUTOUT) {
```

Comment:
Why do we hardcode this behavior for only `CUTOUT`? Would it be better to apply this to any render layer?

Reply:
Currently "no-bloom" rendering shifts anything drawn in `BLOOM` render layer to `CUTOUT` layer. I don't know if there is a need to do this for other layers as well.